Chime/Instructions.m

```diff
@@ -0,0 +1,80 @@
+#include "Headers/Instructions.h"
+#import <Foundation/Foundation.h>
+
+void instruction_op_fetch(NSMapTable* _registers, Stack *_dataStack, NSMutableArray *_ram){
+    NSUInteger valueOfPC = [[_registers objectForKey:@"PC"] integerValue];
+    NSInteger value = [[_ram objectAtIndex:valueOfPC] integerValue];
+    [_dataStack push:@(value)];
+    [_registers setObject:@(valueOfPC + 1) forKey:@"PC"];
+    return;
+}
+
+void instruction_op_load_a(NSMapTable* _registers, Stack *_dataStack, NSMutableArray *_ram){
+    NSUInteger valueOfA = [[_registers objectForKey:@"A"] integerValue];
+    NSInteger value = [[_ram objectAtIndex:valueOfA] integerValue];
+    [_dataStack push:@(value)];
+    return;
+}
+
+void instruction_op_store_a(NSMapTable* _registers, Stack *_dataStack, NSMutableArray *_ram){
+    NSUInteger valueOfA = [[_registers objectForKey:@"A"] integerValue];
+    id value = [_dataStack pop];
+    [_ram setObject:value atIndexedSubscript:valueOfA];
+    return;
+}
+
+void instruction_op_and(Stack *_dataStack){
+    NSInteger value1 = [[_dataStack pop] integerValue];
+    NSInteger value2 = [[_dataStack pop] integerValue];
+    [_dataStack push:@(value1 & value2)];
+    return;
+}
+
+void instruction_op_or(Stack *_dataStack){ NSInteger value1 = [[_dataStack pop] integerValue];
+    NSInteger value2 = [[_dataStack pop] integerValue];
+    [_dataStack push:@(value1 | value2)];
+    return;
+}
+
+void instruction_op_not(Stack *_dataStack){ NSInteger value = [[_dataStack pop] integerValue];
+    [_dataStack push:@(!value)];
```

Comment:
```suggestion
    [_dataStack push:@(~value)];
```
This was my bad, `!` is actually boolean negation. The bitwise not is actually `~`.

Reply:
Also solved in #7